core/src/test/java/org/elasticsearch/cluster/routing/RoutingServiceTests.java

```diff
@@ -138,6 +155,100 @@ public void testDelayedUnassignedScheduleReroute() throws Exception {
         assertThat(routingService.getRegisteredNextDelaySetting(), equalTo(Long.MAX_VALUE));
     }
 
+    /**
+     * This tests that a new delayed reroute is scheduled right after a delayed reroute was run
+     */
+    public void testDelayedUnassignedScheduleRerouteAfterDelayedReroute() throws Exception {
+        final ThreadPool testThreadPool = new ThreadPool(getTestName());
+
+        try {
+            MockClusterService clusterService = new MockClusterService(testThreadPool);
+            DelayedShardsMockGatewayAllocator mockGatewayAllocator = new DelayedShardsMockGatewayAllocator();
+            AllocationService allocation = new AllocationService(Settings.Builder.EMPTY_SETTINGS,
+                    randomAllocationDeciders(Settings.Builder.EMPTY_SETTINGS, new NodeSettingsService(Settings.Builder.EMPTY_SETTINGS), getRandom()),
+                    new ShardsAllocators(Settings.Builder.EMPTY_SETTINGS, mockGatewayAllocator), EmptyClusterInfoService.INSTANCE);
+            RoutingService routingService = new RoutingService(Settings.EMPTY, testThreadPool, clusterService, allocation); // No fake, this is the real deal
+            routingService.start(); // just so performReroute does not prematurely return
+
+            MetaData metaData = MetaData.builder()
+                    .put(IndexMetaData.builder("test1").settings(settings(Version.CURRENT).put(UnassignedInfo.INDEX_DELAYED_NODE_LEFT_TIMEOUT_SETTING, "100ms"))
```

Comment:
can we call these indices  "short_delay" and "long_delay" ? I think it will be easier to read.